server/queues/processors/WebhookProcessor.ts

```diff
@@ -0,0 +1,383 @@
+
+    await this.sendModelWebhook({
+      event,
+      subscription,
+      modelId: event.collectionId,
+      modelPayload: hydratedModel && presentCollection(hydratedModel),
+    });
+  }
+
+  async handleFileOperationEvent(
+    subscription: WebhookSubscription,
+    event: FileOperationEvent
+  ): Promise<void> {
+    const hydratedFileOperation = await FileOperation.findByPk(event.modelId);
+
+    await this.sendModelWebhook({
+      event,
+      subscription,
+      modelId: event.modelId,
+      modelPayload:
+        hydratedFileOperation && presentFileOperation(hydratedFileOperation),
+    });
+  }
+
+  async handleDocumentEvent(
+    subscription: WebhookSubscription,
+    event: DocumentEvent | RevisionEvent
+  ): Promise<void> {
+    const hydratedDocument = await Document.findByPk(event.documentId);
+
+    await this.sendModelWebhook({
+      event,
+      subscription,
+      modelId: event.documentId,
+      modelPayload: hydratedDocument && presentDocument(hydratedDocument),
+    });
+  }
+
+  async handleUserEvent(
+    subscription: WebhookSubscription,
+    event: UserEvent
+  ): Promise<void> {
+    if (event.name === "users.invite") {
+      const invite = event.data;
+
+      await this.sendWebhook({
+        event,
+        subscription,
+        payload: { model: invite },
+      });
+    } else {
+      const hydratedUser = await User.findByPk(event.userId);
+
+      await this.sendModelWebhook({
+        event,
+        subscription,
+        modelId: event.userId,
+        modelPayload: hydratedUser && presentUser(hydratedUser),
+      });
+    }
+  }
+
+  async sendModelWebhook({
+    event,
+    subscription,
+    modelPayload,
+    modelId,
+  }: {
+    event: Event;
+    subscription: WebhookSubscription;
+    modelPayload: unknown;
+    modelId: string;
+  }) {
+    const payload = {
+      id: modelId,
+      model: modelPayload,
+    };
+    await this.sendWebhook({ event, subscription, payload });
+  }
+
+  async sendWebhook({
+    event,
+    subscription,
+    payload,
+  }: {
+    event: Event;
+    subscription: WebhookSubscription;
+    payload: { model: unknown; id?: string };
+  }) {
+    const delivery = await WebhookDelivery.create({
+      webhookSubscriptionId: subscription.id,
+      status: "pending",
+    });
+
+    const jsonBody = presentWebhook({
+      event,
+      delivery,
+      subscription,
+      payload,
+    });
+    const body = JSON.stringify(jsonBody);
+
+    const webhookUrl = subscription.url;
+    const headers = {
+      "Content-Type": "application/json",
+    };
+
+    const response = await fetch(webhookUrl, {
+      method: "POST",
+      headers,
+      body,
+    });
+
+    const newStatus = response.ok ? "success" : "failed";
+
+    await delivery.update({
+      status: newStatus,
+      statusCode: response.status,
+      requestBody: jsonBody,
+      requestHeaders: headers,
+      responseBody: response.body?.toString(),
```

Comment:
This is just saving "[object Object]" in the db, maybe should just change that column to be text instead of blob…

Reply:
Ya I think changing the column type is a good idea, I'll do that!

This actually was saving the contents in the DB correctly, but it wasn't clear why! The `body` property here returns a `ReadbleString` so `toString` actually did the 'correct' thing, at least for what was already buffered.
But I also realized there was an easier way with the fetch api `await response.text()`!'

It also wasn't tested well before but is now!